test/realm_set_test.dart

```diff
@@ -523,6 +523,45 @@ Future<void> main([List<String>? args]) async {
       await Future<void>.delayed(Duration(milliseconds: 20));
     });
 
+    test('RealmSet<$type>.cleared notifications', () async {
```

Comment:
Why not structure this test like for lists and use `expectLater` + `emitsInOrder`? I think it reads better and avoids the brittle delays.

Reply:
I was wondering the same. But decided to keep the tests in the files similar.
OK, I will change it.